components/feature/serviceworker/src/main/java/mozilla/components/feature/serviceworker/ServiceWorkerSupport.kt

```diff
@@ -0,0 +1,51 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.feature.serviceworker
+
+import mozilla.components.concept.engine.Engine
+import mozilla.components.concept.engine.EngineSession
+import mozilla.components.concept.engine.EngineSession.LoadUrlFlags
+import mozilla.components.concept.engine.serviceworker.ServiceWorkerDelegate
+import mozilla.components.feature.tabs.TabsUseCases.AddNewTabUseCase
+import mozilla.components.support.base.log.logger.Logger
+
+/**
+ * Add support for all [Engine] registered service workers' events and callbacks.
+ * Call [install] to actually start observing them.
+ */
+object ServiceWorkerSupport : ServiceWorkerDelegate {
+    private lateinit var engine: Engine
+    private lateinit var addTabUseCase: AddNewTabUseCase
+    private val logger = Logger("ServiceWorkerSupport")
+
+    /**
+     * Start observing service workers' events and callbacks.
+     *
+     * @param engine [Engine] for which to observe service workers events and callbacks.
+     * @param addTabUseCase [AddNewTabUseCase] delegate for opening new tabs when requested by service workers.
+     */
+    fun install(
+        engine: Engine,
+        addTabUseCase: AddNewTabUseCase
+    ) {
+        this.engine = engine
```

Comment:
Indeed, the engine is only used once, in place, in the `install` method.
`addTabUseCase` seems to still be needed to be kept as an internal property though.
If wanting to avoid a `var` for it instead of using `ServiceWorkerSupport.install(<provide all dependencies here>)` we could use a class for something like `ServiceWorkerSupport(<provide all dependencies here>).start()` ? This form is used by Fenix's `WebPushEngineIntegration`.